### ISSUE TITLE
Getting "invalid password" error for `--decrypt` command after upgrading from 8.4.1 to 9.0.2

### ISSUE THREAD

**Issue by ndbroadbent** [qpdf/qpdf#382]
Hello,

I have a test PDF that QPDF `8.4.1` was able to successfully decrypt with `qpdf --decrypt <pdf>`. However, this operation is now failing in `9.0.2`, with the error: `invalid password`. 

This PDF is encrypted, and I don't know the password. However, this was previously working without supplying any password. I assumed that qpdf was still able to remove the password and the encryption. Has this behavior been changed in the latest versions?

QPDF `8.4.1` logs:

```
WARNING: /tmp/20191005-27-z80qpx.pdf (object 1467 0, offset 227378): expected endobj
WARNING: /tmp/20191005-27-z80qpx.pdf (object 1468 0, offset 227664): expected endobj
WARNING: /tmp/20191005-27-z80qpx.pdf (object 1469 0, offset 227950): expected endobj
...
qpdf: operation succeeded with warnings; resulting file may have some problems

# Exit status: 3
```

QPDF `9.0.2` logs:

```
WARNING: /tmp/20191005-27-5rsnag.pdf, object 1472 0 at offset 228700: requested value of integer is too big; returning INT_MAX
/tmp/20191005-27-5rsnag.pdf: invalid password

# Exit status: 2
```

Unfortunately I cannot share this PDF publicly since this is a form that was uploaded by one of my customers. However, I would be happy to share it privately via email. If anyone needs to view this example PDF, then please share your email address, or send me an email at: nathan@formapi.io

Thank you!

**Comment by jberkenbilt** [comment]
If you could email me the file, I will take a look at it. I will delete the file after I've analyzed it. I want to release 9.1.0 soon, and it would be good if whatever this problem is can be fixed for 9.1.

**Comment by jberkenbilt** [comment]
@ndbroadbent I will email you.

**Comment by jberkenbilt** [comment]
Thanks for the file. The file is invalid, but it is just a few lines of code to make qpdf handle it, so I will definitely fix this for 9.1.

If anyone wants to know the details, the encryption dictionary contains a key /P which is supposed to be a negative, signed 32-bit value. This file's P is a positive, unsigned 32-bit value whose bit representation, if interpreted as a 32-bit signed value, is the correct value of /P for this file. Prior to qpdf 9.0.0, qpdf would have silently allowed the unsigned integer to roll over and be treated as a negative value. In > 9.0.0, qpdf detects the overflow. However, in this case, the rollover value is the desired value, and in the specific case of /P, which is a 32-bit bit field, this is harmless since a large value is not going to make it do something like allocate tons of memory or index to a weird place in a file.

My fix is going to be specifically allow /P to be a positive value between INT_MAX and UINT_MAX and to convert it explicitly to a negative value in that case, which would be what qpdf < 9.0.0 was doing by chance.

**Comment by jberkenbilt** [comment]
Note to self: created overloaded QPDFObjectHandle::getIntValueAsInt that allows rollover, and add it to the ABI list to change it to an optional argument.

**Comment by jberkenbilt** [comment]
I added one line of code to test to make sure my fix will work, and with the change, I am able to decrypt the file and get identical results to 8.4.2.

**Comment by ndbroadbent** [comment]
@jberkenbilt Thanks very much, and sorry for sending an invalid PDF! I'm not sure what program generated this PDF, but I'm glad that you were able to spot the issue and find a workaround.

**Comment by jberkenbilt** [comment]
The world is full of invalid PDFs. For most people who write PDF creation software, the only validation they do is to see if Adobe reader opens it. In fact, qpdf started its life as a PDF validator. Over the years, I've seen hundreds of PDF files with all sorts of different kinds of errors, and qpdf has gotten better and better at handling them. It still happens that I get bad PDFs that other readers can open, and it probably always will, but it's also starting to happen that there are PDFs that qpdf can recover so other readers can open them. Anyway, I always like looking at a broken PDF and making qpdf able to read it anyway. :-)

**Comment by jberkenbilt** [comment]
I have coded a fix, which will be in 9.1. This issue will close automatically when the fix merges. Thanks again for the report and for sharing the file. I have deleted the file.
